core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/hashJoin/HashJoinBuildOpExec.scala

```diff
@@ -0,0 +1,57 @@
+package edu.uci.ics.texera.workflow.operators.hashJoin
+
+import edu.uci.ics.amber.engine.architecture.worker.PauseManager
+import edu.uci.ics.amber.engine.common.InputExhausted
+import edu.uci.ics.amber.engine.common.rpc.AsyncRPCClient
+import edu.uci.ics.texera.workflow.common.operators.OperatorExecutor
+import edu.uci.ics.texera.workflow.common.tuple.Tuple
+import edu.uci.ics.texera.workflow.common.tuple.schema.{AttributeType, Schema}
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+
+class HashJoinBuildOpExec[K](
+    val buildAttributeName: String
+) extends OperatorExecutor {
+
+  var buildTableHashMap: mutable.HashMap[K, (ArrayBuffer[Tuple], Boolean)] = _
```

Comment:
For a mutable collection, I am not sure if `ArrayBuffer` or `mutable.List` is better. `ArrayBuffer` seems to have amortized constant append, while `mutable.List` seems to have constant append, according to [this page](https://docs.scala-lang.org/overviews/collections/performance-characteristics.html). Since we are performing appending only, a `mutable.List` might be a better choice?

Reply:
Good point. Changed to `ListBuffer`.